src/ax/marshaling/json_test.go

```diff
@@ -0,0 +1,100 @@
+package marshaling_test
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jmalloc/ax/src/ax/internal/messagetest"
+	. "github.com/jmalloc/ax/src/ax/marshaling"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MarshalJSON", func() {
+
+	message := &messagetest.NonAxMessage{
+		Value: "<value>",
+	}
+
+	It("marshals the message using JSON", func() {
+
+		_, data, err := MarshalJSON(message)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		var m messagetest.NonAxMessage
+		err = json.Unmarshal(data, &m)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(m).Should(Equal(*message))
+	})
+
+	It("includes the protocol information in the content-type", func() {
+		ct, _, err := MarshalJSON(message)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(ct).To(Equal(
+			fmt.Sprintf(
+				"%s; proto=ax.internal.messagetest.NonAxMessage",
+				JSONContentType,
```

Comment:
I think it's best to use a string literal for the content type here (rather than the `JSONContentType` constant), otherwise the tests aren't actually verifying that the function produces the expected output.